emcc.py

```diff
@@ -1400,6 +1400,8 @@ def default_setting(name, new_default):
       shared.Settings.EXPORTED_FUNCTIONS += ['_stbi_load', '_stbi_load_from_memory', '_stbi_image_free']
       # stb_image 2.x need to have STB_IMAGE_IMPLEMENTATION defined to include the implementation
       # when compiling
+      # TODO: Remove this type of linking to stb_image, and instead make this either a system library or a port,
+      # because injecting preprocessing flags causes a dependency to ccache. (bad practice)
```

Comment:
I 100% agree with this!

Reply:
I decided to go ahead and fix this: https://github.com/emscripten-core/emscripten/pull/13658